typescript/packages/plugins/crossmint-headless-checkout/src/crossmint-headless-checkout.plugin.ts

```diff
@@ -0,0 +1,102 @@
+import { Crossmint, CrossmintApiClient, createCrossmint } from "@crossmint/common-sdk-base";
+import { Chain, PluginBase, WalletClientBase, createTool } from "@goat-sdk/core";
+import { z } from "zod";
+
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+
+import { Order } from "@crossmint/client-sdk-base";
+import { parseTransaction } from "viem";
+import packageJson from "../package.json";
+import { getCreateAndPayOrderParameters } from "./parameters";
+
+export class CrossmintHeadlessCheckoutPlugin extends PluginBase {
+    private readonly crossmintApiClient: CrossmintApiClient;
+
+    constructor(
+        private readonly crossmint: Crossmint,
+        private readonly callDataSchema: z.ZodSchema,
```

Comment:
we pass in the callData schema here, as we need to generate the `parameters` zod schema statically - so LLM can know what it needs to pass